tests/integration/test_social_media.py

```diff
@@ -0,0 +1,198 @@
+    assert response_json['posted_at'] == post['posted_at']
+    assert response_json['scheduled_time'] == scheduled_time
+
+
+def test_create_image_post(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'fb',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+    response = client.json.post(url, json.dumps(post))
+
+    assert response.status_code == 201
+    post_id = response.data['id']
+
+    # Request upload image without providing any image
+    url = reverse('posts-upload-image', kwargs={'pk': post_id})
+    response = client.post(url)
+    assert response.status_code == 400
+
+    image = u.create_image(None, 'avatar.png')
+    image = SimpleUploadedFile('front.png', image.getvalue())
+    url = reverse('posts-upload-image', kwargs={'pk': post_id})
+    response = client.post(url, {'image': image}, format='multipart')
+    assert response.status_code == 200
+    assert response.data["image"] is not None
+
+
+def test_delete_image(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'fb',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+    response = client.json.post(url, json.dumps(post))
+
+    assert response.status_code == 201
+    post_id = response.data['id']
+
+    # Delete image before uploading
+    url = reverse('posts-delete-image', kwargs={'pk': post_id})
+    response = client.post(url)
+    assert response.status_code == 400
+
+    image = u.create_image(None, 'avatar.png')
+    image = SimpleUploadedFile('front.png', image.getvalue())
+    url = reverse('posts-upload-image', kwargs={'pk': post_id})
+    response = client.post(url, {'image': image}, format='multipart')
+    assert response.status_code == 200
+    assert response.data["image"] is not None
+
+    url = reverse('posts-delete-image', kwargs={'pk': post_id})
+    response = client.post(url)
+    assert response.status_code == 200
+    assert response.data["image"] is None
+
+
+def test_post_approved_api(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'fb',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+    response = client.json.post(url, json.dumps(post))
+    assert response.status_code == 201
+    post_id = response.data['id']
+
+    super_user = User.objects.create_superuser(email='root@example.com', password='test')
+    client.login(user=super_user)
+    url = reverse('posts-approve', kwargs={'pk': post_id})
+    response = client.post(url)
+    assert response.status_code == 200
+    assert response.data['is_approved']
```

Comment:
This will always be evaluated to `True`. Be explicit about what you want to check here.